src/astx/types/collections.py

```diff
@@ -0,0 +1,98 @@
+"""ASTx collection types module."""
+
+from __future__ import annotations
+
+from typing import List
+
+from public import public
+
+from astx.base import AST, DataType
+from astx.tools.typing import typechecked
+
+
+@public
+@typechecked
+class CollectionType(DataType):
+    """Base class for collection types."""
+
+    pass
+
+
+@public
+@typechecked
+class ListType(CollectionType):
+    """Type representation of a list of elements of specific types."""
+
+    def __init__(self, element_types: List[AST]) -> None:
```

Comment:
about the type, instead of AST, which is very generic, it should be Expr or ExprType (not sure yet which one would fit best here)